### PR TITLE
WT-4351 Ensure resolving prepared transactions use updates from itself

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**
As part of indirect reference resolution in prepared transactions, the transaction operation structure currently always refer the first update in the update chain. This assumption is wrong, first update might be from a different transaction in case of lower isolation levels.

### CODE REVIEW

**Comment by bvpvamsikrishna** [comment]
@sulabhM @agorrod request your review for this pull request.
This is a bug fix in long running prepared transactions.

**Comment by bvpvamsikrishna** [comment]
we are tracking the update count only commit, not for rollback, as our tracking is based on txn id, and incase of rollback, we set it to WT_TXN_ABORTED.

File: src/include/txn.i [hunk 1]
```diff
@@ -359,17 +359,19 @@ __wt_txn_resolve_prepared_op(
 	 * case.
 	 */
 	WT_ASSERT(session, upd != NULL || txn->multi_update_count != 0);
-	if (upd == NULL)
```
Comment (agorrod) [inline src/include/txn.i]: Why not do this check for rollback any more?

**Comment by agorrod** [comment]
Thanks @bvpvamsikrishna lgtm - please merge once testing has passed.

**Pull request merged**